File: 프로그래머스/lv0/120813. 짝수는 싫어요/짝수는 싫어요.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int n) {
    int* answer = (int*)malloc((n+1)/2 * sizeof(int));
    
    for(int i = 1; i <= n; i++){
        if(i%2 == 1)
            answer[i/2] = i;
    }
    return answer;
}

File: 프로그래머스/lv0/120813. 짝수는 싫어요/짝수는 싫어요.java
class Solution {
    public int[] solution(int n) {
        int[] answer = new int [(n+1)/2];
        // 홀수의 경우 n개 이하의 홀수가 배열의 저장되기 위해서는 n을 쓰면 1칸이 부족하게 됨
        for(int i = 1; i <= n; i++){
            if(i % 2 == 1)          // 홀수인 경우
                answer[i/2] = i;
        }
        return answer;
    }
}